src/Input/textfield.css

```diff
@@ -0,0 +1,342 @@
+            right: 0.5rem;
+        }
+    }
+
+    &--dense.d2ui-text-field--with-leading-icon &__input {
+        padding-left: 2.25rem;
+    }
+
+    &--dense.d2ui-text-field--with-leading-icon &__floating-label {
+        left: 2.25rem;
+    }
+
+    &--dense.d2ui-text-field--with-trailing-icon &__input {
+        padding-right: 2.25rem;
+    }
+
+    &--dense.d2ui-text-field--with-value &__floating-label,
+    &--dense &__input:focus ~ &__floating-label {
+        transform: translateY(-65%) scale(0.85);
+    }
+
+    &--dense.d2ui-text-field--outlined.d2ui-text-field--with-value &__floating-label,
+    &--dense.d2ui-text-field--outlined &__input:focus ~ &__floating-label {
+        transform: translateY(-118%) scale(0.85);
+    }
+
+    &--dense.d2ui-text-field--outlined.d2ui-text-field--with-leading-icon.d2ui-text-field--with-value
+        &__floating-label,
+    &--dense.d2ui-text-field--outlined.d2ui-text-field--with-leading-icon
+        &__input:focus
+        ~ &__floating-label {
+        transform: translateY(-118%) translateX(-1.7rem) scale(0.85);
+    }
+
+    /* STATES: VALID */
+    &--valid &__bottom-line {
+        background-color: #1976d2;
+    }
+    &--valid &__notched-outline {
+        border-color: #1976d2;
+    }
+    &--valid.d2ui-text-field--with-value &__floating-label,
+    &--valid &__input:focus ~ &__floating-label,
+    &--valid &__icon--trailing {
+        color: #1976d2;
+    }
+
+    /* STATES: WARNING */
+    &--warning &__bottom-line {
+        background-color: #f19c02;
+    }
+    &--warning &__notched-outline {
+        border-color: #f19c02;
+    }
+    &--warning &__icon--trailing {
+        color: #f19c02;
+    }
+    &--warning.d2ui-text-field--with-value &__floating-label,
+    &--warning &__input:focus ~ &__floating-label {
+        color: #a46a01;
+    }
+
+    /* STATES: DISABLED */
+    &--disabled {
+        cursor: not-allowed;
+    }
+    &--disabled &__input,
+    &--disabled &__floating-label,
+    &--disabled &__icon {
+        color: #9e9e9e;
+        pointer-events: none;
+    }
+    &--disabled.d2ui-text-field--outlined &__input,
+    &--disabled.d2ui-text-field--outlined &__input:hover,
+    &--disabled.d2ui-text-field--minimal &__input,
+    &--disabled.d2ui-text-field--minimal &__input:hover {
+        border-color: rgba(196, 196, 196, 0.2);
+    }
+    &--disabled.d2ui-text-field--filled &__input:hover {
+        border-color: transparent;
+        /* Darken 5% */
+        background-color: #eceff1;
+    }
+
+    /* STATES: ERROR */
+    &--error &__bottom-line {
+        background-color: #e53935;
+    }
+    &--error &__notched-outline {
+        border-color: #e53935;
+    }
+    &--error.d2ui-text-field--with-value &__floating-label,
+    &--error &__input:focus ~ &__floating-label,
+    &--error &__icon--trailing {
+        color: #e53935;
+    }
+}
+/* IE11 hacks */
+@media all and (-ms-high-contrast: none) {
+    .d2ui-text-field {
+        border-radius: 0;
```

Comment:
Again... If you disable this line and open the example app in IE11 you will see some white pixels in the rounded corners. Maybe you know of a way to get rid of them.